test/unittests/t_libcvmfs.cc

```diff
@@ -423,34 +425,33 @@ TEST_F(T_Libcvmfs, StatNestedCatalog) {
   EXPECT_EQ(LIBCVMFS_ERR_OK,
     cvmfs_attach_repo_v2((tester.repo_name().c_str()), opts, &ctx));
 
-  const char *orig_nc = NULL;
-
   // stat nested catalog using client repo
   struct cvmfs_nc_attr *nc_attr;
   nc_attr = cvmfs_nc_attr_init();
   EXPECT_FALSE(cvmfs_stat_nc(ctx, "/dir/dir/dir/dir", nc_attr));
   EXPECT_FALSE(strcmp(d4_hash, nc_attr->hash));
   EXPECT_FALSE(strcmp("/dir/dir/dir/dir", nc_attr->mountpoint));
   cvmfs_nc_attr_free(nc_attr);
-
+  free(d4_hash);
 
   // stat nested catalog using client repo
   nc_attr = cvmfs_nc_attr_init();
   EXPECT_FALSE(cvmfs_stat_nc(ctx, "/dir", nc_attr));
-  orig_nc = tester.manifest()->catalog_hash().ToString().c_str();
-  EXPECT_FALSE(strcmp(orig_nc, nc_attr->hash));
+  EXPECT_FALSE(strcmp(d0_hash, nc_attr->hash));
   EXPECT_FALSE(strcmp("", nc_attr->mountpoint));
   // Size of root catalog is 0 as a Nested Catalog
-  EXPECT_EQ(nc_attr->size, 0);
+  size_t expected = 0;
+  EXPECT_EQ(nc_attr->size, expected);
```

Comment:
`EXPECT_EQ(0U, nc_attr->size)` -- GTest wants the expected value first for proper reporting when it fails.